server/datastore/mysql/software.go

```diff
@@ -276,26 +301,23 @@ func listSoftwareDB(
 		)
 	}
 
-	sql, args, err := ds.ToSQL()
-	if err != nil {
-		return nil, ctxerr.Wrap(ctx, err, "sql build")
-	}
-
-	var result []fleet.Software
-	if err := sqlx.SelectContext(ctx, q, &result, sql, args...); err != nil {
-		return nil, ctxerr.Wrap(ctx, err, "load host software")
-	}
+	return ds.ToSQL()
+}
 
-	if opts.SkipLoadingCVEs {
-		return result, nil
+func countSoftwareDB(
+	ctx context.Context, q sqlx.QueryerContext, hostID *uint, opts fleet.SoftwareListOptions,
+) (int, error) {
+	opts.ListOptions = fleet.ListOptions{
```

Comment:
Noop?

Reply:
Nvm, now I get it, to make it less confusing API-wise I'd define an options struct for software count only...